application/src/main/java/org/togetherjava/tjbot/logging/discord/DiscordLogForwarder.java

```diff
@@ -199,12 +198,17 @@ private static String describeLogEvent(LogEvent event) {
             return logMessage + "\n" + exceptionWriter.toString().replace("\t", "> ");
         }
 
-        private static @Nullable String linkToSource(String source, String sourceCodeBaseUrl) {
+        private static Optional<String> linkToSource(String source, String sourceCodeBaseUrl) {
             if (!source.startsWith(BASE_PACKAGE)) {
-                return null;
+                return Optional.empty();
+            }
+
+            if (!sourceCodeBaseUrl.endsWith("/")) {
+                sourceCodeBaseUrl += "/";
             }
 
-            return sourceCodeBaseUrl + source.replace('.', '/') + ".java";
+            String link = sourceCodeBaseUrl + source.replace('.', '/') + ".java";
```

Comment:
`String link = "%s%s.java".formatted(sourceCodeBaseUrl, source.replace('.', '/'));` to avoid the String concat currently being performed.